conans/client/cmd/new.py

```diff
@@ -145,6 +153,9 @@ def source(self):
 
     def package(self):
         self.copy("*.h", "include")
+
+    def package_id(self):
+        self.info.header_only()
```

Comment:
I understand this is more explicit than just a recipe without settings, but we are adding more code here that is unneeded. If this is a pattern followed in CCI for example, then I am ok with it. Maybe we should consider it for conan 2.0 (enforce settings always?)

Reply:
I guess it might be error prone without explicit, e.g. if you add some settings and options to the recipe after running `conan new`, it's easy to forget about this method.
same if you decide to add more requirements, as `header_only` now contains:
```
self.requires.clear()
```
I believe the same package id is expected to be generated regardless of requires, which is not the case in the current template.